src/common/cockpitloopback.c

```diff
@@ -66,16 +66,15 @@ cockpit_loopback_next_async (GSocketAddressEnumerator *enumerator,
                              gpointer user_data)
 {
   CockpitLoopback *self = COCKPIT_LOOPBACK (enumerator);
-  GSimpleAsyncResult *res;
+  GTask *res;
   GSocketAddress *address;
   GError *error = NULL;
 
   address = cockpit_loopback_next (enumerator, cancellable, &error);
   g_assert (error == NULL);
 
-  res = g_simple_async_result_new (G_OBJECT (self), callback, user_data, NULL);
-  g_simple_async_result_set_op_res_gpointer (res, address, address ? g_object_unref : NULL);
-  g_simple_async_result_complete_in_idle (res);
+  res = g_task_new (G_OBJECT (self), NULL, callback, user_data);
+  g_task_return_pointer (res, address, address ? g_object_unref : NULL);
```

Comment:
minor nag: GTask already contains a `NULL` check on the destroy notify, so you don't need to conditionalise the callback

Reply:
Convenient! Simplified.